### PR TITLE
react-frontend: Generate gql types

### PR DESCRIPTION
**Pull request by allisonbraithw**
This adds frontend type generation.

### CODE REVIEW

File: react-frontend/codegen.ts [hunk 1]
```diff
@@ -0,0 +1,15 @@
+import { CodegenConfig } from '@graphql-codegen/cli'
+
+const config: CodegenConfig = {
+    schema: '../gateway/supergraph.graphql',
+    documents: ['src/**/*.tsx'],
+    ignoreNoDocuments: true,
+    generates: {
+        './src/gql/': {
```
Comment (tmc) [inline react-frontend/codegen.ts]: should this be named something that indicates it's' generated?
separately, a .gitattributes file indicating this path is generated would be helpful, see https://github.com/tmc/hackathon-starter/blob/master/go-graphql-backend/.gitattributes for inspo.

File: react-frontend/package.json [hunk 2]
```diff
@@ -3,17 +3,18 @@
   "version": "0.1.0",
   "private": true,
   "dependencies": {
+    "@apollo/client": "^3.7.14",
```
Comment (tmc) [inline react-frontend/package.json:6]: should we use apollo client or urql? I like urql's simplicity and have heard some stories around apollo's caching complexity. Thoughts?

**Review by tmc — APPROVED** [review]
LGTM

File: react-frontend/src/App.tsx [hunk 3]
```diff
@@ -10,7 +10,6 @@ const getUserQueryDocument = graphql(/* GraphQL */`
   query GetUser($userId: ID!) {
     user(id: $userId) {
       id
-      username
```
Comment (tmc) [inline react-frontend/src/App.tsx]: I'd like this sample query to hit both backends

File: react-frontend/codegen.ts (hunk 1 above)
Comment (tmc) [reply react-frontend/codegen.ts]: gql-generated?

File: react-frontend/src/App.tsx (hunk 3 above)
Comment (allisonbraithw) [reply react-frontend/src/App.tsx]: same

File: react-frontend/package.json (hunk 2 above)
Comment (allisonbraithw) [reply react-frontend/package.json:6]: I don't have a strong preference, happy to try both

**Pull request merged**